pkg/certificateauthority/caconfigmap.go

```diff
@@ -56,9 +57,21 @@ func MountCaConfigMap(deployment *appsv1.Deployment) {
 // getCaOnDisk returns the contents of the Certificate Authority bundle on disk as a byte
 // array or returns the error encountered when attempting to do so.
 func getCaOnDisk() ([]byte, error) {
-	b, err := ioutil.ReadFile(trustedCaMountPath + caBundlePath)
+	return getFileOnDisk(trustedCaMountPath + caBundlePath)
+}
+
+// getCaOnDisk returns the contents of the Certificate Authority bundle on disk as a byte
+// array or returns the error encountered when attempting to do so.
+func getFileOnDisk(path string) ([]byte, error) {
+	b, err := ioutil.ReadFile(path)
 	if err != nil {
 		return []byte{}, err
 	}
 	return b, nil
 }
+
+// isRunningOnPod returns true if marketplace is being ran on a pod.
+func isRunningOnPod() bool {
+	_, err := getFileOnDisk("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
Nit: `getFileOnDisk` is responsible for reading the content of a file, using it to check the existence of a file seems to be a misuse. Maybe ` info, err := os.Stat(filename)` is more appropriate?

Reply:
Fair point - I'll make the switch.